src/main.cpp

```diff
@@ -224,7 +237,8 @@ static void frame_handle_buffer(void *, struct zwlr_screencopy_frame_v1 *frame,
     if (buffer.height % 2)
         buffer.height -= 1;
 
-    if (!buffer.wl_buffer) {
+    if (!buffer.wl_buffer || buffer.format != format) {
```

Comment:
`buffer.format != format`, but `buffer.format = (wl_shm_format)format;` at line 228.
So I added `uint32_t buffer_format = buffer.format;` at line 228.
Changed current line to `if (!buffer.wl_buffer || buffer_format != format) {`， 
And then it started working properly.